gui/gui.py

```diff
@@ -0,0 +1,517 @@
+import logging
+# PyQt needs to be imported like this because for whatever reason they decided not to include a __all__ = [...]
+import PyQt5.QtWidgets as QtWidgets
+import PyQt5.QtGui as QtGui
+
+from config import config
+from model import DirectoryStatus
+
+from .config import ConfigDialog
+from . import tools, widgets, colours, threads
+
+import os
+
+
+class MainDialog(QtWidgets.QDialog):
+    def __init__(self):
+        super().__init__()
+        logging.debug("GUI: Setting up MainDialog")
+
+        self.config_dlg = ConfigDialog(parent=self)
+
+        # set window title and icon
+        self.setWindowTitle("IRIDA Uploader")
+        icon_path = os.path.join(os.path.dirname(__file__), 'images/icon.ico')
+        self.setWindowIcon(QtGui.QIcon(icon_path))
+
+        # Initialize threads from the tools file
+        self._status_thread = threads.StatusThread()
+        self._parse_thread = threads.ParseThread()
+        self._upload_thread = threads.UploadThread()
```

Comment:
I assume these are GUI threads which get cleaned up by the gui framework on close?  Anything specific need added for error states?

Reply:
The framework does all the cleanup stuff for us. From all the examples I looked at I don't think there is any need for additional error handling.